### PR TITLE
 Integrate BERT into Hedwig

### PR DESCRIPTION
**Pull request by achyudh**

### CODE REVIEW

File: datasets/imdb.py
Comment (daemon) [inline datasets/imdb.py:43]: This change doesn't break previous code, right?

File: datasets/reuters.py
Comment (daemon) [inline datasets/reuters.py:57]: Same here.

File: datasets/sst.py
Comment (daemon) [inline datasets/sst.py:23]: List of integers? Floats?

File: models/distill/__main__.py
Comment (daemon) [inline models/distill/__main__.py]: Unnecessary comments.

File: utils/tokenization.py
Comment (daemon) [inline utils/tokenization.py:71]: Did we modify this class in any way? We can just add `pytorch_pretrained_bert` to the requirements and import `BertTokenizer`, right?

File: utils/tokenization.py
Comment (daemon) [inline utils/tokenization.py]: `strip()` isn't necessary for a `split()` with no args.

**Review by daemon — CHANGES_REQUESTED** [review]
Nothing major. In the future, we should move toward the recommended state dictionary serialization scheme.

File: utils/tokenization.py
Comment (achyudh) [reply utils/tokenization.py:71]: I would prefer having it in our code rather than adding an additional dependency. It also makes future modifications of the tokenizer easy.  @Ashutosh and I were also thinking about trying Word Piece tokens on our existing models to see how they perform.

File: datasets/imdb.py
Comment (achyudh) [reply datasets/imdb.py:43]: It doesn't break previous code. Directory structure is different (in a way simpler) in Hedwig when compared to Castor.

**Comment by daemon** [comment]
`./models/bert/model.py:            shutil.rmtree(tempdir)`
This clears everything in the saves directory?

**Comment by daemon** [comment]
Maybe not, but for whatever reason my Yelp model file is gone after the Reuters run.

**Comment by achyudh** [comment]
@daemon Please don't merge this pull request until issue #31 is fixed. I'm working on it.

**Comment by achyudh** [comment]
@Ashutosh-Adhikari before we merge this into master, can you please address issue #32?

File: models/distill/__main__.py
Comment (achyudh) [reply models/distill/__main__.py]: This file was removed in 0cdb70ac3322a7187ae5443b641fd44709aef60a

File: datasets/reuters.py
Comment (achyudh) [reply datasets/reuters.py:57]: It doesn't break previous code. Directory structure is different (in a way simpler) in Hedwig when compared to Castor.

File: utils/tokenization.py
Comment (achyudh) [reply utils/tokenization.py]: Fixed in 870a2ec28e89f0e16d81b4fed83ec4ed09181b35

**Comment by achyudh** [comment]
Both issues #31 and #32 have been resolved. @daemon can you please review these changes and merge them if there are no issues?

**Review by daemon — APPROVED** [review]

**Pull request merged**
